src/Files.App/ViewModels/HomeViewModel.cs

```diff
@@ -8,34 +8,92 @@ namespace Files.App.ViewModels
 {
 	public sealed class HomeViewModel : ObservableObject, IDisposable
 	{
-		public ObservableCollection<WidgetContainerItem> WidgetItems { get; } = new();
+		// Dependency injections
+
+		private IUserSettingsService UserSettingsService { get; } = Ioc.Default.GetRequiredService<IUserSettingsService>();
+
+		// Properties
+
+		public ObservableCollection<WidgetContainerItem> WidgetItems { get; } = [];
+
+		// Commands
 
 		public ICommand HomePageLoadedCommand { get; }
 
-		public event EventHandler<RoutedEventArgs>? HomePageLoadedInvoked;
-		public event EventHandler? WidgetListRefreshRequestedInvoked;
+		// Constructor
 
 		public HomeViewModel()
 		{
 			HomePageLoadedCommand = new RelayCommand<RoutedEventArgs>(ExecuteHomePageLoadedCommand);
 		}
 
-		private void ExecuteHomePageLoadedCommand(RoutedEventArgs? e)
+		// Methods
+
+		public void ReloadWidgets()
 		{
-			HomePageLoadedInvoked?.Invoke(this, e!);
+			var reloadQuickAccessWidget = WidgetsHelpers.TryGetWidget<QuickAccessWidgetViewModel>(this);
+			var reloadDrivesWidget = WidgetsHelpers.TryGetWidget<DrivesWidgetViewModel>(this);
+			var reloadFileTagsWidget = WidgetsHelpers.TryGetWidget<FileTagsWidgetViewModel>(this);
+			var reloadRecentFilesWidget = WidgetsHelpers.TryGetWidget<RecentFilesWidgetViewModel>(this);
+
+			if (reloadQuickAccessWidget)
```

Comment:
Yeah WidgetHelper send to me a bool that indicates whether the app should load xaml of the widget(if loaded, viewmodel will be constructed). Unless, never loaded.